cms/lib/xblock/student_properties/student_properties.py

```diff
@@ -48,5 +48,14 @@ def get_event_context(self, event_type, event):  # pylint: disable=unused-argume
                     result['registration'][prop.name] = prop.value
                 elif prop.course_id and str(self.runtime.course_id) == str(prop.course_id):
                     result['enrollment'][prop.name] = prop.value
+
+            for item in user.credomodulesuserprofile_set.filter(course_id=self.runtime.course_id):
```

Comment:
I suggest to use such construction:
```
profile = CredoModulesUserProfile.objects.get(user=user, course_id=course_key)
```
also it will be good to add new method to the `CredoModulesUserProfile` class to return `meta` `dict` (I mean to hide all convert from `json` to `dict` logic inside the class method)